lib/fs.js

```diff
@@ -2551,459 +2275,3 @@ Object.defineProperty(fs, 'SyncWriteStream', {
-  },
-
-  async readdir(path, options) {
-    options = getOptions(options, {});
-    path = getPathFromURL(path);
-    validatePath(path);
-    return binding.readdir(pathModule.toNamespacedPath(path),
-                           options.encoding, kUsePromises);
-  },
-
-  async readlink(path, options) {
-    options = getOptions(options, {});
-    path = getPathFromURL(path);
-    validatePath(path, 'oldPath');
-    return binding.readlink(pathModule.toNamespacedPath(path),
-                            options.encoding, kUsePromises);
-  },
-
-  async symlink(target, path, type_) {
-    const type = (typeof type_ === 'string' ? type_ : null);
-    target = getPathFromURL(target);
-    path = getPathFromURL(path);
-    validatePath(target, 'target');
-    validatePath(path);
-    return binding.symlink(preprocessSymlinkDestination(target, type, path),
-                           pathModule.toNamespacedPath(path),
-                           stringToSymlinkType(type),
-                           kUsePromises);
-  },
-
-  async fstat(handle) {
-    validateFileHandle(handle);
-    return statsFromValues(await binding.fstat(handle.fd, kUsePromises));
-  },
-
-  async lstat(path) {
-    path = getPathFromURL(path);
-    validatePath(path);
-    return statsFromValues(
-      await binding.lstat(pathModule.toNamespacedPath(path), kUsePromises));
-  },
-
-  async stat(path) {
-    path = getPathFromURL(path);
-    validatePath(path);
-    return statsFromValues(
-      await binding.stat(pathModule.toNamespacedPath(path), kUsePromises));
-  },
-
-  async link(existingPath, newPath) {
-    existingPath = getPathFromURL(existingPath);
-    newPath = getPathFromURL(newPath);
-    validatePath(existingPath, 'existingPath');
-    validatePath(newPath, 'newPath');
-    return binding.link(pathModule.toNamespacedPath(existingPath),
-                        pathModule.toNamespacedPath(newPath),
-                        kUsePromises);
-  },
-
-  async unlink(path) {
-    path = getPathFromURL(path);
-    validatePath(path);
-    return binding.unlink(pathModule.toNamespacedPath(path), kUsePromises);
-  },
-
-  async fchmod(handle, mode) {
-    mode = modeNum(mode);
-    validateFileHandle(handle);
-    validateUint32(mode, 'mode');
-    if (mode < 0 || mode > 0o777)
-      throw new errors.RangeError('ERR_OUT_OF_RANGE', 'mode');
-    return binding.fchmod(handle.fd, mode, kUsePromises);
-  },
-
-  async chmod(path, mode) {
-    path = getPathFromURL(path);
-    validatePath(path);
-    mode = modeNum(mode);
-    validateUint32(mode, 'mode');
-    return binding.chmod(pathModule.toNamespacedPath(path), mode, kUsePromises);
-  },
-
-  async lchmod(path, mode) {
-    if (constants.O_SYMLINK !== undefined) {
-      const fd = await promises.open(path,
-                                     constants.O_WRONLY | constants.O_SYMLINK);
-      return promises.fschmod(fd, mode).finally(fd.close.bind(fd));
```

Comment:
this is the bug caught by moving out of an object

Reply:
@targos mind explaining why catching the bug required moving it out of an object?

Reply:
AFAIK eslint cannot check properties on objects. It is not something that can be easily done statically.
The best way to catch such errors is to increase code coverage.